shuup/core/api/service.py

```diff
@@ -10,14 +10,43 @@
 from parler_rest.serializers import (
     TranslatableModelSerializer, TranslatedFieldsField
 )
+from rest_framework.fields import empty
 from rest_framework.serializers import SerializerMethodField
 
-from shuup.core.models import PaymentMethod, ShippingMethod
+from shuup.core.models import (
+    PaymentMethod, ServiceBehaviorComponent, ShippingMethod
+)
+
+
+class DynamicBehaviorComponentSerializer(TranslatableModelSerializer):
+
+    class Meta:
+        fields = ("__all__")
+
+    def __init__(self, instance=None, data=empty, **kwargs):
+        self.Meta.model = instance.__class__
+        super(DynamicBehaviorComponentSerializer, self).__init__(instance, data, **kwargs)
+
+
+class BehaviorComponentSerializer(TranslatableModelSerializer):
+    class Meta:
+        model = ServiceBehaviorComponent
+        fields = ("__all__")
+
+    def to_representation(self, obj):
+        """
+        Behavior Component representator
+
+        BehaviorComponents are Polymorphic thus it's required to make
+        the `to_representation` trick.
+        """
+        return DynamicBehaviorComponentSerializer(obj, context=self.context).to_representation(obj)
 
 
 class ServiceBaseSerializer(TranslatableModelSerializer):
     price = SerializerMethodField()
     is_available = SerializerMethodField()
+    behavior_components = SerializerMethodField()
```

Comment:
I don't like this idea since this serializer is used by public API. Also not sure whether there is real need for this.